mephisto/abstractions/providers/mturk/wrap_crowd_source.js

```diff
@@ -15,6 +15,22 @@ Returning None for the assignment_id means that the task is being
 previewed by the given worker.
 \------------------------------------------*/
 
+let eventEmitter = () => ({
+  events: {},
+  emit(event, ...args) {
+    let callbacks = this.events[event] || [];
+    for (let i = 0, length = callbacks.length; i < length; i++) {
+      callbacks[i](...args);
+    }
+  },
+  on(event, cb) {
+    this.events[event]?.push(cb) || (this.events[event] = [cb]);
```

Comment:
just realized something. this file gets loaded in directly into the HTML page that runs on the client's browser: https://github.com/facebookresearch/Mephisto/blob/210e407e0683ce4cef273325746d07cc82d62fc4/examples/static_react_task_with_tips/webapp/src/static/index.html#L22

As such, we maaaay run into some slight compat issues with "newer" ES6 syntax such as the spread syntax and optional chaining "?."

- https://caniuse.com/mdn-javascript_operators_optional_chaining
- https://caniuse.com/mdn-javascript_operators_spread

I would suggest using the [Babel transpiler](https://babeljs.io/repl#?browsers=defaults&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=DYUwLgBCBuIHZgKIFsCWYwgE4QLwQAoBKPAPkIG8AoCKWBAZwC4IKBfAGhqjTAJnhgOEAHRiAhlgDmDEtVq1QkAMbjgwAEbjlAawZ4IYABaoGIgYwDaFsAF0IAHwcRLtgNzdaAMwD2OAkoQqAYADMKgcFLGBqrqWrpmEVFGbkEQADwQScapqADUeXKeCrGa2nqWqLYEYiKSMkQeChBs3JzcPnD89EIQyhpFzcam5j0M1j22APwiAA4ArgxGBP0kToTDZjbjNvb4lv22jcVY4PNYcIQkuOTyzYYmW2MTgnsPI9svCNMiXqjAmCwBAIqGu5H6EAAhLh8KCmgo2PD2mxGkA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Creact%2Cstage-3&prettier=false&targets=Node-10.2&version=7.18.10&externalPlugins=&assumptions=%7B%7D) to make sure we use a `target` with more coverage. I suggest playing around with the `targets` value in the sidebar here:

![CleanShot 2022-08-03 at 10 35 48@2x](https://user-images.githubusercontent.com/425059/182635878-caf077fb-7744-40e8-8056-ef83267ebab3.png)

Reply:
Let's also then minify the transpiled output above into maybe a single line of compressed output and then add that to the file. We can then leave a comment above it explaining our process for generating it nanoevents lib -> transpiler -> minifer